### PR TITLE
Port fs commands to the new shell

### LINKED ISSUE #8874: Adapt fs subsystem to new shell subsystem
FS is widely using existing shell and it needs to be ported to the new one once it is ready.

### PR DESCRIPTION
**Pull request by nordic-krch**
Post shell fs commands to the new shell.

It has changes on top of #9362 

Fixes #8874.

### CODE REVIEW

File: subsys/fs/shell.c [hunk 1]
```diff
@@ -177,34 +305,32 @@ static int cmd_read(int argc, char *argv[])
...
+		shell_fprintf(shell, SHELL_NORMAL, " ");
 
 		for (i = 0; i < read; i++) {
-			printk("%c", buf[i] < 32 ||
+			shell_fprintf(shell, SHELL_NORMAL, "%c", buf[i] < 32 ||
 			       buf[i] > 127 ? '.' : buf[i]);
 		}
 
-		printk("\n");
+		shell_fprintf(shell, SHELL_NORMAL, "\n");
```
Comment (jarz-nordic) [inline subsys/fs/shell.c]: I think it shall be `\r\n`

File: subsys/fs/shell.c [hunk 2]
```diff
@@ -72,49 +65,181 @@ static void create_abs_path(const char *name, char *path, size_t len)
...
 
 	create_abs_path(argv[1], path, sizeof(path));
 
 	res = fs_mkdir(path);
 	if (res) {
-		printk("Error creating dir[%d]\n", res);
-		return res;
+		shell_fprintf(shell, SHELL_ERROR,
+				"Error creating dir[%d]\r\n", res);
+		return;
```
Comment (jarz-nordic) [inline subsys/fs/shell.c]: obsolete return

**Comment by nordic-krch** [comment]
fyi @nvlsianpu can you take a look at this?

**Review by jakub-uC — APPROVED** [review]

**Comment by nvlsianpu** [comment]
sure

**Review by nashif — APPROVED** [review]

File: subsys/fs/shell.c [hunk 3]
```diff
@@ -147,20 +265,24 @@ static int cmd_read(int argc, char *argv[])
 
 	err = fs_stat(path, &dirent);
 	if (err) {
-		printk("Failed to stat %s (%d)\n", path, err);
-		return err;
+		shell_fprintf(shell, SHELL_ERROR, "Failed to stat %s (%d)\r\n",
```
Comment (nvlsianpu) [inline subsys/fs/shell.c]: error message: "Failed to obtain/get file/file-syetem object status"

Comment (nvlsianpu) [reply subsys/fs/shell.c]: I understand that message was inherited :D

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: subsys/fs/shell.c (hunk 3 above)
Comment (nordic-krch) [reply subsys/fs/shell.c]: yes, i didn't touch messages but i fixed that one on your request :)

**Comment by jarz-nordic** [comment]
@nordic-krch : May you plesae rebase to this PR:#10322?

**Review by jakub-uC — CHANGES_REQUESTED** [review]
Please rebase to PR#10322.

**Comment by nordic-krch** [comment]
rebased on master (including #10322

**Review by jakub-uC — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**